DataFormats/TrackerRecHit2D/interface/OmniClusterRef.h

```diff
@@ -7,28 +7,32 @@
 #include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
 #include "DataFormats/Phase2TrackerCluster/interface/Phase2TrackerCluster1D.h"
 #include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/FTLRecHit/interface/FTLClusterCollections.h"
 
 class OmniClusterRef {
 
-  static const unsigned int kInvalid = 0x80000000; // bit 31 on
-  static const unsigned int kIsStrip = 0x20000000; // bit 29 on
+  static const unsigned int kInvalid  = 0x80000000; // bit 31 on
+  static const unsigned int kIsStrip  = 0x20000000; // bit 29 on
   // FIXME:: need to check when introducing phase2 pixel 
   static const unsigned int kIsPhase2 = 0x40000000; // bit 30 on
+  static const unsigned int kIsTiming = 0x10000000; // bit 28 on
   static const unsigned int kIsRegional = 0x60000000; // bit 30 and 29 on  (will become fastsim???)
-
-  static const unsigned int indexMask = 0xFFFFFF;
-  static const unsigned	int subClusMask = 0xF000000;
+  
+  static const unsigned int indexMask   =   0xFFFFFF;
+  static const unsigned	int subClusMask =  0xF000000;
   static const unsigned int subClusShift = 24;
 
 public:
   typedef edm::Ref<edmNew::DetSetVector<SiPixelCluster>,SiPixelCluster > ClusterPixelRef;
   typedef edm::Ref<edmNew::DetSetVector<SiStripCluster>,SiStripCluster > ClusterStripRef;
   typedef edm::Ref<edmNew::DetSetVector<Phase2TrackerCluster1D>, Phase2TrackerCluster1D> Phase2Cluster1DRef;
+  typedef edm::Ref<FTLClusterCollection,FTLCluster> ClusterMTDRef;
   
   OmniClusterRef() : me(edm::RefCore(),kInvalid) {}
   explicit OmniClusterRef(ClusterPixelRef const & ref, unsigned int subClus=0) : me(ref.refCore(), (ref.isNonnull() ? ref.key()                  | (subClus<<subClusShift) : kInvalid) ){  }
   explicit OmniClusterRef(ClusterStripRef const & ref, unsigned int subClus=0) : me(ref.refCore(), (ref.isNonnull() ? (ref.key()   | kIsStrip  ) | (subClus<<subClusShift) : kInvalid) ){ }
   explicit OmniClusterRef(Phase2Cluster1DRef const & ref, unsigned int subClus=0) : me(ref.refCore(), (ref.isNonnull() ? (ref.key() | kIsPhase2) | (subClus<<subClusShift) : kInvalid) ){ }
+  explicit OmniClusterRef(ClusterMTDRef const & ref) : me(ref.refCore(), (ref.isNonnull() ? (ref.key() | kIsPhase2 | kIsTiming) : kInvalid) ){  }
```

Comment:
`kTiming` should be treated as all other cases instead of a sub-case of `kIsPhase2`. The only meaning of the enumerator is to point to the concrete type, and `kIsPhase` corresponds to `Phase2Cluster1DRef`.
```suggestion
  explicit OmniClusterRef(ClusterMTDRef const & ref) : me(ref.refCore(), (ref.isNonnull() ? (ref.key() | kIsTiming) : kInvalid) ){  }
```